geth/jail/internal/loop/loop_test.go

```diff
@@ -0,0 +1,56 @@
+package loop_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/status-im/status-go/geth/jail/internal/loop/looptask"
+
+	"github.com/robertkrimen/otto"
+	"github.com/status-im/status-go/geth/jail/internal/loop"
+	"github.com/status-im/status-go/geth/jail/internal/vm"
+	"github.com/stretchr/testify/suite"
+)
+
+func (s *LoopSuite) TestAddAndReady() {
+	t := looptask.NewIdleTask()
+
+	err := s.loop.Add(t)
+	s.NoError(err)
+
+	err = s.loop.Ready(t)
+	s.NoError(err)
+
+	s.cancel()
+
+	// Wait for the context to cancel and loop to close
+	time.Sleep(100 * time.Millisecond)
+
+	err = s.loop.Add(t)
+	s.Error(err)
+
+	err = s.loop.Ready(t)
+	s.Error(err)
+}
+
+type LoopSuite struct {
+	suite.Suite
+
+	loop   *loop.Loop
+	cancel context.CancelFunc
+}
+
+func (s *LoopSuite) SetupTest() {
```

Comment:
Hmm, seems more general:

- Standard test function on top
- Then the suite type definition
- Then setting up and tear down of suite and tests
- And after that all suite tests

Not yet sure if we defined it as standard, but seen in `e2e` tests and it seems logical.

Reply:
I think I understand what you mean and have changed the order.

Reply:
I believe the functions in `loop_test.go` have been reordered to match this request.